text/0000-pack-publish-buildpack.md

```diff
@@ -0,0 +1,287 @@
+
+1. Opens a link to a new Github Issue in the user's browser
+    - The user must be logged into Github
+    - The link will use [Github query parameters](https://help.github.com/en/github/managing-your-work-on-github/about-automation-for-issues-and-pull-requests-with-query-parameters) to pre-populate the title, body, and labels.
+    - The issue will be opened against the `https://github.com/buildpacks/registry-index` repo.
+    - The issue body will contain structured data that defines the buildpack id and version to yank.
+1. A Github action will detect the new issue and do the following:
+    - If the issue is in an unexpected format, or if the digest does not match the image located at `url-or-tag`, the bot will add a comment to the issue and close it.
+    - Confirm that the Github user who opened the issue is an owner of the buildpack.
+    - Create and merge a PR against the `https://github.com/buildpacks/registry-index` repo using a Gitub token (i.e. all commits in that repo will be made by the same "user"). The PR will set `yanked=true` for the buildpack version described in the issue.
+    - Close the Github issue.
+
+The following behavior will execute when the configured registry is of type `git`.
+
+1. Creates a `local` branch on the local registry index cache
+1. Makes a Git commit to the local registry index with an update setting `yank=true` in the JSON payload representing the buildpack version as defined in [RFC-0022](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md)
+1. Uses Git to push the commit to the registry index's `origin`.
+1. Updates the `master` branch of the local registry cache from `origin`.
+
+Options:
+
+* `--buildpack-registry, -R` - the id of a registry configured in `.pack/config.toml`
+* `--undo` - this option will execute the same flow, but set `yanked=false`
+
+Example:
+
+```
+$ pack yank-buildpack buildpacks/sample@0.0.1
+```
+
+## Future Work
+
+Not included in this proposal, but intended for future RFCs are the following:
+
+* `pack add-buildpack-registry` - a new command that will add to the `[[registries]]` config in `.pack/config.toml`.
+* `pack set-buildpack-registry` - a new command that will mutate the `default-registry` config in `.pack/config.toml`.
+* `pack publish-buildpack` - a new command that will perform both `package-buildpack` and `register-buildpack` in one step.
+* *Headless release* - a new option on `register-buildpack` would allow for a non-interactive (i.e. no browser) registration
+* *REST API* - as an alternative to `github` and `git` types of registries, we intend to support an `api` type that will use a registry service to broker registration requests (instead of using Github Issues).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* `type` of registry isn't accounted for in the [Buildpack URI](https://github.com/buildpacks/rfcs/blob/master/text/0037-buildpack-uris.md) definition
```

Comment:
I'm failing to see where this is relevant? In what context would `type` be necessary in a UR(L|I)?

Reply:
@jromero what i mean here is that I'm not sure what those `cnb://` URLs really look like. Is it `cnb://github.com/buildpacks/registry-index/<ns>/<name>`? How do I reference a CNB on other Git servers?

Reply:
Ah got it. This makes sense. Sounds like a superseding RFC to come for CNB URIs.